src/maker/api.rs

```diff
@@ -147,39 +148,51 @@ impl Maker {
             data_dir.unwrap_or(get_maker_dir())
         };
 
-        let wallet_dir = data_dir.join("wallets");
+        let wallets_dir = data_dir.join("wallets");
 
         let mut rpc_config = rpc_config.unwrap_or_default();
 
-        // Load/Create wallet depending on if a wallet with wallet_file_name exists.
-        let mut wallet = if let Some(file_name) = wallet_file_name {
-            let wallet_path = wallet_dir.join(&file_name);
-            rpc_config.wallet_name = file_name;
-            if wallet_path.exists() {
-                // Try loading wallet
-                let wallet = Wallet::load(&rpc_config, &wallet_path)?;
-                log::info!("Wallet file at {:?} successfully loaded.", wallet_path);
-                wallet
-            } else {
-                // Create wallet with the given name.
-                let mnemonic = Mnemonic::generate(12).map_err(WalletError::BIP39)?;
-                let seedphrase = mnemonic.to_string();
+        let mnemonic = Mnemonic::generate(12).unwrap();
+        let seedphrase = mnemonic.to_string();
 
-                let wallet = Wallet::init(&wallet_path, &rpc_config, seedphrase, "".to_string())?;
-                log::info!("New Wallet created at : {:?}", wallet_path);
-                wallet
-            }
+        // Derive the wallet path based on the provided `wallet_file_name` or by scanning the `wallets_dir`.
+        // If `wallet_file_name` is provided, join it with `wallets_dir` to form the path.
+        // If `wallet_file_name` is not provided, check if a wallet file already exists in `wallets_dir`:
+        // - If a file exists, return its path.
+        // - If no file exists, generate a default file name and return the constructed path.
+        let wallet_path = if let Some(file_name) = wallet_file_name {
+            wallets_dir.join(&file_name)
         } else {
-            // Create default wallet
-            let mnemonic = Mnemonic::generate(12).map_err(WalletError::BIP39)?;
-            let seedphrase = mnemonic.to_string();
-
-            // File names are unique for default wallets
-            let unique_id = seed_phrase_to_unique_id(&seedphrase);
-            let file_name = unique_id + "-maker";
-            let wallet_path = wallet_dir.join(&file_name);
-            rpc_config.wallet_name = file_name;
+            fs::read_dir(&wallets_dir)
+                .ok()
+                .and_then(|read_dir| {
+                    read_dir
+                        .filter_map(|entry| entry.ok().map(|e| e.path()))
+                        .find(|path| path.is_file())
```

Comment:
This will return the first file it finds in the dir. So if there are multiple files then it will return the first find. Which can be any of them, so that's pretty confusing.

What it can do instead is: try to find a file with a `deafult-name`. If that exists then load it, if not, then create a new file with the `default-name`.

This is different from our current default nameing behavior, which creates a unique name for each new wallet, but then we cannot simulate any "default loading" behavior. I couldn't think of any other way than to add the `deafult_name` concept. This is how core also does it so it's pretty standard. The flow goes like this:

- Theres a `default_name` hard coded. Lets say `taker-wallet`.
- if `wallet-file_name == None`, check if default wallet exists in the dir.
- If yes, then load it, or else create a new wallet with the `deafult_name`.

We call this the `deafult_wallet`. And if user never specifies any wallet name, it will always use this wallet.

Few restrictions we need to maintain at user's side:
- there can never be multiple wallets with the `default_name`. 
- If the user wants to create a new wallet with the default name, it will error saying "default wallet already exists, if you wanna create a new wallet choose a new name".

As far as I remember, we added the random naming idea for the tests, and they might fail withoutthis. But in theory they should not, as the datadir for each makers are different.

Lets fix this whole problem together in this PR including test fixes if any.